src/differentiable_vectors/autodiff.jl

```diff
@@ -0,0 +1,42 @@
+# if only f! and vector shaped x_seed is given, construct j! or fj! according to autodiff
+function DifferentiableVector(f!, x_seed::AbstractVector; J = nothing, autodiff=:central)
+    if autodiff == :central
+        function fj!(fx, jx, x)
+            f!(fx, x)
+            function f(x)
+                fx = similar(x)
+                f!(fx, x)
+                return fx
+            end
+            finite_difference_jacobian!(f, x, fx, jx, autodiff)
+        end
+        function j!(jx, x)
```

Comment:
If they aren't going to be handled now, you should open an issue about handling these allocations better. Is there a reason you don't just pre-allocate and enclose the `fx` here instead of getting a new one every time??

Reply:
I didn't really look at this code besides renaming stuff an making sure it didn't work. I was waiting for DiffEqDiffTools to clean this up.